File: frontend/src/types/job.ts
export type JobStatus = 'queued' | 'running' | 'pause_requested' | 'paused' | 'completed' | 'failed';

export interface JobAcceptedResponse {
  job_id: number;
  status: JobStatus;
}

export interface BackgroundJob {
  id: number;
  job_type: string;
  status: JobStatus;
  result: Record<string, any> | null;
  error: string | null;
  created_at: string;
  started_at: string | null;
  finished_at: string | null;
}

File: backend/app/schemas/job.py
from __future__ import annotations

import datetime
from typing import Any, Optional

from pydantic import BaseModel


class JobAcceptedResponse(BaseModel):
    job_id: int
    status: str


class JobResponse(BaseModel):
    id: int
    job_type: str
    status: str
    result: Optional[dict[str, Any]] = None
    error: Optional[str] = None
    created_at: datetime.datetime
    started_at: Optional[datetime.datetime] = None
    finished_at: Optional[datetime.datetime] = None

